src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -1146,11 +1146,62 @@ subroutine Initialize(gc, import, export, clock, rc)
          rc=status                                                              &
          )
     _VERIFY(status)
-    
+
     call ESMF_TimeIntervalSet(LandAssim_DT, s=LandAssimDtStep, rc=status)
     _VERIFY(status)
+    
+    _ASSERT(mod(LandAssimDtStep,model_dtstep)==0, "inconsistent inputs for HEARTBEAT_DT and LANDASSIM_DTSTEP")
 
+    call MAPL_DateStampGet( clock, datestamp, rc=status)
+    _VERIFY(STATUS)
+    read(datestamp(1:8),*)   nymd
+    read(datestamp(10:13),*) nhms
+    nhms = nhms*100
+    
+    ! read reference time for land analysis from resource file (default: 0z)
+    call MAPL_GetResource(                                                      &
+         MAPL,                                                                  &
+         landassim_reftime,                                                     &
+         'LANDASSIM_REFTIME:',                                                  &
+         default=0,                                                             &
+         rc=status                                                              &
+         )
+    _VERIFY(status)
 
+    ! determine date and time of first land analysis   [R20210506: not sure I understand the intent here correctly.] 
+    !
+    ! RR20210506: Not sure this is right. Users should be able to specify any "sensible" reftime.
+    !             That is, the reftime should not need to change when the initial (restart) time changes.
+    !             e.g., reftime=0z,    initial time=0z, landassim_dt=3h  --> first analysis is at 3z    (next at 6z, 9z, ...)
+    !             e.g., reftime=3z,    initial time=0z, landassim_dt=3h  --> first analysis is at 3z
+    !             e.g., reftime=0z,    initial time=3z, landassim_dt=3h  --> first analysis is at 6z
+    !             e.g., reftime=1:30z, initial time=0z, landassim_dt=3h  --> first analysis is at 1:30z (next at 4:30z, ...)
+    !             e.g., reftime=7:30z, initial time=0z, landassim_dt=3h  --> first analysis is at 1:30z (next at 4:30z, ...)
+    ! If AssimTime is indeed meant to be the time of the first land analysis, I'm pretty sure hour/min/sec of AssimTime
+    !   would need to depend on "current" time (ie, "datestamp") as well as "reftime"
+    ! Put differently, "landassim_reftime" and "landassim_dtstep" effectively define an infinite sequence of land analysis time, something like:
+    !   ..., [some date] 0z, 3z, 6z, 9z, ..., 21z, [next day], 0z, 3z, ... 21z, [another day later] 0z, 3z, ... 21z, [yet another day later] 0z, 3z, ...
```

Comment:
Ok, my ref time is absolute time. Yours is relative to start time. I am fine with your definition

Reply:
If it is relative to start time. Some of my calculation needs to be changed